csharp_dotnetcore/3.Welcome-Users/Startup.cs

```diff
@@ -0,0 +1,135 @@
+        /// Initializes a new instance of the <see cref="Startup"/> class.
+        /// ASP.NET Core Application Statup code. This method gets called by the runtime.
+        /// Use this method to add services to the container.
+        /// For more information on how to configure your application, visit <see ref="https://docs.microsoft.com/en-us/aspnet/core/fundamentals/startup"/>.
+        /// </summary>
+        /// <param name="env">Provides information about the web hosting environment an application is running in.</param>
+        public Startup(IHostingEnvironment env)
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+        }
+
+        /// <summary>
+        /// Gets the configuration that represents a set of key/value application configuration properties.
+        /// </summary>
+        /// <value>
+        /// The IConfiguration that represents a set of key/value application configuration properties.
+        /// </value>
+        public IConfiguration Configuration { get; }
+
+        /// <summary>
+        /// This method gets called by ASP.NET Core runtime as part of initializaing your application
+        /// Use this method to add services to the container.
+        /// </summary>
+        /// <param name="services">Services.</param>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            try
+            {
+                services.AddBot<WelcomeUserBot>(options =>
+                {
+                    options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+
+                    // The Memory Storage used here is for local bot debugging only. When the bot
+                    // is restarted, anything stored in memory will be gone.
+                    IStorage dataStore = new MemoryStorage();
+
+                    // For production bots use Azure Blob, or Azure CosmosDB storage provides
+                    // as seen below. To include any of the Azure based storage providers,
+                    // add the Microsoft.Bot.Builder.Azure  Nuget package to your solution. That package is found at:
+                    //      https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/
+
+                    // IStorage Store = new Microsoft.Bot.Builder.Azure.AzureBlobStorage("AzureBlobConnectionString", "containerName");
+
+                    // var convoState = new ConversationState(dataStore);
+                    // options.State.Add(new ConversationState(dataStore));
+                    options.State.Add(new UserState(dataStore));
+
+                    // Add State to BotStateSet Middleware (that require auto-save)
```

Comment:
@daveta & I moved his samples to a pattern that didn't need BotStateSet, and instead called save directly. This resulted in less code and was more clear to read. Please sync with @daveta to adopt that same pattern here.